src/config/config.go

```diff
@@ -65,40 +69,59 @@ func ProcessArgs() *Config {
 	flag.StringVar(&ret.HTTPSProxy, "https-proxy", "", "A proxy URL to use for creating HTTPS connections outside the cluster")
 	flag.StringVar(&ret.NoProxy, "no-proxy", "", "A comma-separated list of destination domain names, domains, IP addresses, or other network CIDRs to exclude proxying")
 	flag.StringVar(&ret.ServiceIPs, "service-ips", "", "All IPs of assisted service node")
-	flag.StringVar(&ret.HighAvailabilityMode, "high-availability-mode", "Full", "high-availability expectations. default is 'Full', which represents the behavior in a \"normal\" cluster. Use 'None' for single-node deployment")
+	flag.StringVar(&ret.HighAvailabilityMode, "high-availability-mode", "", "high-availability expectations, \"Full\" which represents the behavior in a \"normal\" cluster. Use 'None' for single-node deployment. Leave this value as \"\" for workers as we do not care about HA mode for workers.")
 	flag.BoolVar(&ret.CheckClusterVersion, "check-cluster-version", false, "Do not monitor CVO")
 	flag.StringVar(&ret.MustGatherImage, "must-gather-image", "", "Custom must-gather image")
 	flag.Var(&ret.DisksToFormat, "format-disk", "Disk to format. Can be specified multiple times")
 
 	var installerArgs string
 	flag.StringVar(&installerArgs, "installer-args", "", "JSON array of additional coreos-installer arguments")
-
 	h := flag.Bool("help", false, "Help message")
 	flag.Parse()
 
+	if err := ret.SetInstallerArgs(installerArgs); err != nil {
+		printHelpAndExit(err)
+	}
+
+	if h != nil && *h {
+		printHelpAndExit(nil)
+	}
+
 	if ret.NoProxy != "" {
 		utils.SetNoProxyEnv(ret.NoProxy)
 	}
-	if h != nil && *h {
-		printHelpAndExit()
+
+	ret.SetDefaults()
+	if err := ret.Validate(); err != nil {
+		printHelpAndExit(err)
 	}
+	return ret
+}
 
+func (c *Config) SetInstallerArgs(installerArgs string) error {
 	if installerArgs != "" {
-		err := json.Unmarshal([]byte(installerArgs), &ret.InstallerArgs)
-		if err != nil {
-			println(err.Error())
-			printHelpAndExit()
-		}
+		return json.Unmarshal([]byte(installerArgs), &c.InstallerArgs)
 	}
-	if ret.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+	return nil
+}
+
+func (c *Config) Validate() error {
+	if c.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
 		validRoles := []string{string(models.HostRoleMaster), string(models.HostRoleBootstrap)}
-		if !funk.ContainsString(validRoles, ret.Role) {
-			println("high-availability-mode is set to None, but host role is %s. should be one of: %s", ret.Role, validRoles)
-			printHelpAndExit()
+		if !funk.ContainsString(validRoles, c.Role) {
+			return fmt.Errorf("high-availability-mode is set to None, but host role is %s. should be one of: %s", c.Role, validRoles)
 		}
 	}
-	if ret.InfraEnvID == "" {
-		ret.InfraEnvID = ret.ClusterID
+	return nil
+}
+
+func (c *Config) SetDefaults() {
+	if c.HighAvailabilityMode != "" && c.Role == string(models.HostRoleWorker) {
```

Comment:
not sure i understand why we are doing it here. Change agent code and don't send it in case it is worker

Reply:
after discussion i understand why it is needed. 
As for me there is no need to check c.HighAvailabilityMode != ""